test/dynamo/test_higher_order_ops.py

```diff
@@ -429,7 +876,7 @@ def false_fn(x):
         res = mod_for_compile(torch.tensor(True), torch.tensor(5))
         res = mod_for_compile(torch.tensor(True), torch.tensor(5))
 
-        self.assertEqual(len(backend.graphs), 2)
+        self.assertEqual(len(backend.graphs), 0)
```

Comment:
hmmm why did this become 0?

Reply:
There's an append to nonlocal list in cond's true_fn. Since this is a side effect disallowed by this PR, we fall back on the entire cond to eager-mode PyTorch. The thing that is being compiled is just the cond, so there are zero graphs.